modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/CreateAnAPIApplicationEndpoint.testcase

```diff
@@ -93,6 +116,291 @@ definition {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("When I click 'Add API Endpoint' button") {
+			Click(locator1 = "APIBuilder#ADD_API_ENDPOINT");
+		}
+
+		task ("Then I can see path under {portalUrl}/o/c/{applicationName}") {
+			var portalURL = JSONCompany.getPortalURL();
+
+			AssertTextEquals(
+				locator1 = "APIBuilder#URL_LABEL",
+				value1 = "${portalURL}/o/c/test2");
+		}
+
+		task ("And Then I can see path start with '/' with placeholder 'Enter path'") {
+			AssertElementPresent(locator1 = "APIBuilder#SLASH");
+		}
+	}
+
+	@priority = 4
+	test CanSeeScopeOfScopeFields {
+		property portal.acceptance = "true";
+
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("When I click 'ADD API Endpoint' button") {
+			Click(locator1 = "APIBuilder#ADD_API_ENDPOINT");
+		}
+
+		task ("Then I click Scope option") {
+			Click(
+				key_value = "Select Scope",
+				locator1 = "APIBuilder#DROPDOWN_TOGGLE");
+		}
+
+		task ("And Then I can see site and company scope in the Scope option") {
+			AssertElementPresent(
+				key_optionName = "Site",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+
+			AssertElementPresent(
+				key_optionName = "Company",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+		}
+	}
+
+	@priority = 4
+	test CanSeeValidationMessageForRequiredFields {
+		property portal.acceptance = "true";
+
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I click 'Add API Endpoint' button") {
+			Click(locator1 = "APIBuilder#ADD_API_ENDPOINT");
+		}
+
+		task ("When I click 'Create' button in 'New API Endpoint'") {
+			Click(locator1 = "APIBuilder#CREATE_BUTTON");
+		}
+
+		task ("Then I can see error message about required Scope and Path") {
+			AssertElementPresent(
+				key_alertMessage = "Please select a scope.",
+				locator1 = "Message#ERROR_ENTER_A_VALID_VALUE");
+
+			AssertElementPresent(
+				key_alertMessage = "Please enter a path.",
+				locator1 = "Message#ERROR_ENTER_A_VALID_VALUE");
+		}
+
+		task ("And Then no endpoint is being created") {
+			Click(locator1 = "Modal#CLOSE_BUTTON");
+
+			AssertElementPresent(
+				key_entryName = "No API Endpoint Found",
+				locator1 = "ObjectAdmin#VIEW_ENTRY_NAME_ON_PAGE");
+		}
+	}
+
+	@priority = 3
+	test CanValidateMaximumDescription {
+		task ("And Given I go to 'Endpoints' tab in 'Edit API Application'") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
+				entity = "Endpoints",
+				title = "test2");
+		}
+
+		task ("And Given I enter a path and description Field with 281 characters") {
+			APIBuilderUI.createAPIEndpoint(
```

Comment:
In `APIBuilderUI.createAPIEndpoint` we already click the `Create` button, there is no need to click it again in When step, we can update 
`task ("And Given I enter a path and description Field with 281 characters") {`
to be 
`task ("When I create an endpoint with description Field contains 281 characters") {`
and remove the 
`task ("When I Click 'Create' button") {`
Once you updated the code here, then our qa headless will update in our test scenarios, thanks

Reply:
Done